### PR TITLE
[New] `no-absolute-path`: add fixer

### PR DESCRIPTION
**Pull request by adipascu**
Hello, this is my first contribution to this repo. Should I change anything?

### CODE REVIEW

**Comment by adipascu** [comment]
After reading https://github.com/nodejs/node/issues/6049 I think I should have used [path.posix](https://nodejs.org/api/path.html#pathposix).
Are any of these tests ran on MS Windows?

**Comment by ljharb** [comment]
Yes, the travis tests are on Mac and the appveyor ones on Windows.

File: src/rules/no-absolute-path.js [hunk 1]
```diff
@@ -10,13 +11,25 @@ module.exports = {
...
-        context.report(source, 'Do not import modules using an absolute path');
+        context.report({
+          node: source,
+          message: 'Do not import modules using an absolute path',
+          fix: fixer => {
+            const resolvedContext = context.getPhysicalFilename ? context.getPhysicalFilename() : context.getFilename();
+            let relativePath = path.relative(path.dirname(resolvedContext), source.value);
+            if (!relativePath.startsWith('.')){
+              relativePath = '.' + path.sep + relativePath;
+            }
```
Comment (ljharb) [inline src/rules/no-absolute-path.js:29]:
```suggestion
            if (!relativePath.startsWith('.')) {
              relativePath = path.join('.', relativePath);
            }
```

Comment (adipascu) [reply src/rules/no-absolute-path.js]: `path.join('.', relativePath)` drops the leading `.`, this is why I chose to do explicit string concatenation.

**Comment by adipascu** [comment]
I've updated the PR to always do posix style paths.
- This seems to be [the way node.js works in general](https://github.com/nodejs/node/issues/6049)
 - [Browser URLs](https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname) seem to only support `/` posix separators
 I hope this will pass windows CI.

**Comment by adipascu** [comment]
I've updated the PR to always do posix style paths.
- This seems to be [the way node.js works in general](https://github.com/nodejs/node/issues/6049)
 - [Browser URLs](https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname) seem to only support `/` posix separators
 I hope this will pass windows CI.
 Edit: sorry for the force pushes, should I do separate commits with any feedback/follow up work?

**Comment by ljharb** [comment]
Force pushes are always fine; if i'm curious about the push diff i can click on "force pushed" in the github timeline :-)

File: src/rules/no-absolute-path.js (hunk 1 above)
Comment (ljharb) [reply src/rules/no-absolute-path.js]: ah ok, fair enough

File: src/rules/no-absolute-path.js [hunk 2]
```diff
@@ -10,13 +11,26 @@ module.exports = {
...
       if (isAbsolute(source.value)) {
-        context.report(source, 'Do not import modules using an absolute path');
+        context.report({
+          node: source,
+          message: 'Do not import modules using an absolute path',
+          fix: fixer => {
+            const resolvedContext = context.getPhysicalFilename ? context.getPhysicalFilename() : context.getFilename();
+            // node.js and web imports work with posix style paths ("/")
+            let relativePath = path.posix.relative(path.dirname(resolvedContext), source.value);
+            if (!relativePath.startsWith('.')){
```
Comment (ljharb) [inline src/rules/no-absolute-path.js:28]:
```suggestion
            if (!relativePath.startsWith('.')) {
```

File: tests/src/rules/no-absolute-path.js [hunk 3]
```diff
@@ -53,48 +55,76 @@ ruleTester.run('no-absolute-path', rule, {
...
+      output: 'var f = require("../../some/path")',
     }),
     // validate amd
     test({
       code: 'require(["/some/path"], function (f) { /* ... */ })',
+      filename: '/foo/bar/index.js',
       options: [{ amd: true }],
       errors: [error],
+      output:
+        'require(["../../some/path"], function (f) { /* ... */ })',
```
Comment (ljharb) [inline tests/src/rules/no-absolute-path.js:119]:
```suggestion
      output: 'require(["../../some/path"], function (f) { /* ... */ })',
```
i smell prettier :-p

File: tests/src/rules/no-absolute-path.js [hunk 4]
```diff
@@ -53,48 +55,76 @@ ruleTester.run('no-absolute-path', rule, {
...
     }),
     test({
       code: 'define(["/some/path"], function (f) { /* ... */ })',
+      filename: '/foo/bar/index.js',
       options: [{ amd: true }],
       errors: [error],
+      output:
+        'define(["../../some/path"], function (f) { /* ... */ })',
```
Comment (ljharb) [inline tests/src/rules/no-absolute-path.js:127]:
```suggestion
      output: 'define(["../../some/path"], function (f) { /* ... */ })',
```

**Review by ljharb — APPROVED** [review]
looks good if tests pass

**Comment by adipascu** [comment]
I've accepted the white-space suggestions (and they made separate commits).

I hope this gets squash merged.

Thank you for being receptive to my changes!

**Comment by adipascu** [comment]
CI had network issues, I've triggered a fresh round.

**Comment by adipascu** [comment]
Hi @ljharb, could we merge this in?

**Pull request merged**